### DISCUSSION TITLE
[Help] Would it be possible to render a note template and only return the note content as a string?

### DISCUSSION THREAD

**Discussion by mpas** [zk-org/zk-nvim#140] [Help]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

I am looking for a way to place a piece of content into an existing note in the background. 

Something like:

```
-- insert content below a markdown_header
function InsertContent(filename, markdown_header, content) ... end
```

For the content I would like to render a template using zk-nvim and looking for a way to just get the rendered text and provide that rendered text to the function above..

I looked at the API but unsure if the rendering of a template and just getting the string result is available without creating a new note file. I am aware of the `inline` option for my guess is that for this the document/note should be open.

**Comment by mickael-menu** [comment]
You can use the `dryRun` option with the `zk.new` LSP command, and it will return the body of the generated note without actually creating it.

https://github.com/mickael-menu/zk/blob/main/docs/editors-integration.md#zknew

**Reply by mpas** [reply]
I am trying to find an example but I am not successful in using the zk.new command. I tried the following simple example to generate the content but the result of the print is nil

```
  local zk = require("zk")
  local options = { template = "todo.md", content = "dummy content", dryRun = true }
  local c = zk.new(options)
  print(c)
```

**Reply by mpas** [reply]
I guess that the way I am trying to achieve what I want is not going to work. I am trying to put the code to get some template rendered in a non buffer attached way. So i created a simple function that should return the rendered template but instead it is returning nil due to the fact that the LSP server is not attached.

```
function AddTodo()
  local zk = require("zk")
  local options = { template = "todo.md", content = "dummy content", dryRun = true }
  local c = zk.new(options)
  print(c)
end

```

**Reply by mickael-menu** [reply]
The `zk.new` Lua function opens the note after creating it. It's a high level user function.

You want to use the `zk.api.new` function which takes a callback function which will contain the content of the note. As shown here: https://github.com/mickael-menu/zk-nvim/blob/797618aa07f58ceba6f79fb6e777e8e45c51e1ce/lua/zk.lua#L67-L72

(the first parameter can be nil)

**Comment by mpas** [comment]
Worked wonderfull! I am now able to use a custom command, ask for user input and save a todo in my notes files that stores alls my todos. This for quickly adding TODOs if needed!! Thanks a lot for the help. Learned a lot.

Not sure if my Lua code is ok but at least it works :) Happy me!
